src/server/index.test.js

```diff
@@ -67,19 +68,21 @@ test('sync', () => {
   // Sync causes the server to respond.
   expect(io.socket.emit).toHaveBeenCalledTimes(0);
   io.socket.receive('sync', 'gameID');
-  expect(io.socket.emit).toHaveBeenCalledTimes(1);
```

Comment:
Wrapping this assertion inside a timeout works fine but I think there should be a better solution for `async` behavior of `socket` callbacks

Reply:
Just use `await` in the tests. The trick is to make the function explicitly have a `return` statement. Then you can use `await` to wait for its execution and then test what it did. In this case, I wrapped `receive` in this way to make the tests to work.